src/reconcile/rec_write.c

```diff
@@ -2672,7 +2681,8 @@ __rec_copy_prev_addr(WT_SESSION_IMPL *session, WTI_RECONCILE *r)
         WT_ASSERT_ALWAYS(session, false, "write delta for a new page.");
         break;
     case WT_PM_REC_REPLACE: /* 1-for-1 page swap */
-        multi->block_meta = page->block_meta;
+        if (page->disagg_info != NULL)
```

Comment:
This function is called only in the delta case. As we are adding the check, it would be good add in the below case also while accessing the `multi`. Otherwise, remove it both the cases and add an assert at the start of the function.

Reply:
fixed